src/profile-logic/process-profile.js

```diff
@@ -1052,56 +1056,195 @@ export function processProfile(
+  thread: Thread,
+  threadIndex: number,
+  stringArray: string[],
+  PIIToBeRemoved: PII
+): Thread | null {
+  if (
+    PIIToBeRemoved.hiddenThreads !== null &&
+    PIIToBeRemoved.hiddenThreads.includes(threadIndex)
+  ) {
+    // If this is a hidden thread, remove the thread immediately.
+    // This will not remove the thread entry from the `threads` array right now
+    // and just replace it with a `null` value. We filter out the null values
+    // inside `serializeProfile` function.
+    return null;
+  }
+
+  // We iterate all the markers and remove/change data depending on the PII
+  // status.
+  const markersToDelete = [];
+  if (
+    PIIToBeRemoved.networkUrls ||
+    PIIToBeRemoved.screenshots === true ||
+    (Array.isArray(PIIToBeRemoved.screenshots) &&
+      PIIToBeRemoved.screenshots.length > 0)
+  ) {
+    for (let i = 0; i < thread.markers.length; i++) {
+      const currentMarker = thread.markers.data[i];
+
+      // Remove the all network URLs if user wants to remove them.
+      if (
+        PIIToBeRemoved.networkUrls &&
+        currentMarker &&
+        currentMarker.type &&
+        currentMarker.type === 'Network'
+      ) {
+        // Remove the URI fields from marker payload.
+        currentMarker.URI = '';
+        currentMarker.RedirectURI = '';
+        // Strip the URL from the marker name
+        const stringIndex = thread.markers.name[i];
+        stringArray[stringIndex] = stringArray[stringIndex].replace(/:.*/, '');
+      }
+
+      // If PIIToBeRemoved.screenshots is true, this means that we need to
+      // remove all screenshots, if it's an array of thread IDs, that means we
+      // should only delete screenshots from the given threads.
+      if (
+        (PIIToBeRemoved.screenshots === true ||
+          (Array.isArray(PIIToBeRemoved.screenshots) &&
+            PIIToBeRemoved.screenshots.includes(threadIndex))) &&
+        currentMarker &&
+        currentMarker.type &&
+        currentMarker.type === 'CompositorScreenshot'
+      ) {
+        const urlIndex = currentMarker.url;
+        stringArray[urlIndex] = '';
+        markersToDelete.push(i);
+      }
+    }
+  }
+
+  // After iterating (or not iterating at all) the markers, if we have some
+  // markers we want to delete or user wants to delete the full time range,
+  // reconstruct the marker table and samples table without unwanted information.
+  if (markersToDelete.length > 0 || PIIToBeRemoved.fullTimeRange !== null) {
+    const oldMarkers = thread.markers;
+    const newMarkerTable = getEmptyRawMarkerTable();
+    const addCurrentMarker = (index: number) => {
+      if (markersToDelete.includes(index)) {
+        return;
+      }
+      newMarkerTable.name.push(oldMarkers.name[index]);
+      newMarkerTable.time.push(oldMarkers.time[index]);
+      newMarkerTable.data.push(oldMarkers.data[index]);
+      newMarkerTable.length++;
+    };
+
+    // If user wants to remove full time range, filter all the markers and
```

Comment:
"Full time range" is a new term. I believe this is referring to the "committed time range". I'm mainly pointing this out since the time ranges have proved difficult to keep straight, so it's nice to have consistent terms. There are a lot of uses of this term here, and elsewhere in the patch.

Reply:
No, this is referring to "root range". I think the term here means "should we remove everything but the committed range". Maybe the boolean should be something like `shouldRemoveInvisibleRange` ?

Reply:
bikeshed: `shouldFilterToCommittedRange`? `invisibleRange` is still a bit unclear to me.